x-pack/plugins/secops/public/components/timeline/body/column_headers/header.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as fp from 'lodash/fp';
+import * as React from 'react';
+import { pure } from 'recompose';
+import { OnColumnSorted, OnFilterChange } from '../../events';
+import { Sort, SortDirection } from '../sort/sort';
+import { SortIndicator } from '../sort/sort_indicator';
+import { ColumnHeader } from './column_header';
+import { Filter } from './filter';
+
+interface GetSortDirectionParams {
+  header: ColumnHeader;
+  sort: Sort;
+}
+
+/** Given a `header`, returns the `SortDirection` applicable to it */
+export const getSortDirection = ({ header, sort }: GetSortDirectionParams): SortDirection =>
+  header.id === sort.columnId ? sort.sortDirection : 'none';
+
+/** Given a current sort direction, it returns the next sort direction */
+export const getNextSortDirection = (currentSort: Sort): SortDirection => {
+  switch (currentSort.sortDirection) {
+    case 'descending':
+      return 'ascending';
+    case 'ascending':
+      return 'descending';
+    default:
```

Comment:
This seems kind of off to me.

Isn't the code for SortDirection only 3 different possible states?:

```
export type SortDirection = 'none' | 'ascending' | 'descending';
```

So, shouldn't this be a case statement of `none` and then it is to return 'descending' to rotate around correctly. `default` would really be a programmer error or fatal condition where you should never be able to get to the code or the programmer added a 4th state but didn't know where to plumb it?

Like if I was writing this in another language I think it would force the only 3 conditions with `default` not being able to exist?